schemas/expense.json

```diff
@@ -0,0 +1,102 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://schema.kosmos.org/Expense",
+  "type": "object",
+  "title": "Expense",
+  "description": "Record of costs expended",
+  "default": {},
+  "required": [
+    "title",
+    "currency",
+    "amount"
+  ],
+  "additionalProperties": false,
+  "properties": {
+    "@context": {
+      "type": "string",
+      "enum": [ "https://schema.kosmos.org" ],
+      "default": "https://schema.kosmos.org"
+    },
+    "@type": {
+      "type": "string",
+      "enum": [ "Expense" ],
+      "default": "Expense"
+    },
+    "title": {
+      "type": "string",
+      "title": "Title",
+      "description": "A short description of the expense",
+      "default": "",
+      "examples": [
+        "Server rent"
+      ]
+    },
+    "description": {
+      "type": "string",
+      "title": "Description",
+      "description": "A longer description of the expense",
+      "default": "",
+      "examples": [
+        "Dedicated server: andromeda.kosmos.org"
+      ]
+    },
+    "currency": {
+      "type": "string",
+      "title": "Currency",
+      "description": "ISO 4217 (or other) code of the currency that the expense was paid with",
+      "default": "",
+      "examples": [
+        "EUR",
+        "USD",
+        "BTC"
+      ]
+    },
+    "amount": {
+      "type": "number",
+      "title": "Amount",
+      "description": "Expended amount of defined currency",
+      "default": 0,
+      "examples": [
+        42.0
```

Comment:
JSON does not have integer values. You can either use numbers or strings. Most use cases for currency use the number type in my experience.

Reply:
We do not know the currency or its amount of decimals here. I would rather see actual BTC than Satoshis for example, because otherwise you have to implement conversions for looking at all the various different ones properly. And you would have to specify an additional unit, not just the currency code, for it to be correct.

Reply:
we would not need to know anything. We would simply say it is in the smallest denomination - kinda the same as we do with the human-every-day used denomination. And there are actually currencies where that's not so clear, my favorit here Iran: https://en.wikipedia.org/wiki/Iranian_toman

ideally you'd specify a unit either way. In any documentation I am always wondering about the unit of an "amount" field.

I also think calculating with integer values is  easier.

but I don't have strong feelings here. I think it's fine.

Reply:
(re conversion: I guess that's a bit of the question if the raw document is for humans or computers. 
When we create a ETH/BTC transaction we have to do a conversion then. )

Reply:
The unit of 1 EUR, USD, BTC, DAI, ETH etc. is one full unit of account. I think 0.1 EUR is as clear as 0.1 ETH or 0.1 BTC, whereas with Lightning and BTC for example, the base unit could easily change from sats to millisats in the future, So you would have to specify that it is indeed Satoshis you're using, while with the full unit and decimals, you simply use the correct amount as per base unit (1 sat being 0.00000001 BTC and 1 msat being 0.00000000001). See e.g. https://en.bitcoin.it/wiki/Units#Table_of_all_units